internal/quickfort/place.lua

```diff
@@ -137,32 +166,35 @@ end
 
 local function init_containers(db_entry, ntiles, fields)
     if db_entry.want_barrels then
-        local max_barrels =
+        local max_barrels = db_entry.num_barrels or
                 quickfort_common.settings['stockpiles_max_barrels'].value
         if max_barrels < 0 or max_barrels >= ntiles then
             fields.max_barrels = ntiles
         else
             fields.max_barrels = max_barrels
         end
+        log('barrels set to %d', fields.max_barrels)
     end
     if db_entry.want_bins then
-        local max_bins = quickfort_common.settings['stockpiles_max_bins'].value
+        local max_bins = db_entry.num_bins or
+                quickfort_common.settings['stockpiles_max_bins'].value
         if max_bins < 0 or max_bins >= ntiles then
             fields.max_bins = ntiles
         else
             fields.max_bins = max_bins
         end
+        log('bins set to %d', fields.max_bins)
     end
-    if db_entry.want_wheelbarrows then
-        local max_wb =
+    if db_entry.want_wheelbarrows or db_entry.num_wheelbarrows then
+        local max_wb = db_entry.num_wheelbarrows or
                 quickfort_common.settings['stockpiles_max_wheelbarrows'].value
-        if max_wb < 0 then
-            fields.max_wheelbarrows = 1
-        elseif max_wb >= ntiles then
-            fields.max_wheelbarrows = ntiles
+        if max_wb < 0 then max_wb = 1 end
+        if max_wb >= ntiles - 1 then
+            fields.max_wheelbarrows = ntiles - 1
```

Comment:
I think this is the case, but just to confirm - you are directly setting this field, not sending keypresses to the game, right? Asking because `tweak max-wheelbarrows` changes the in-game wheelbarrow UI (the vanilla UI uses `w` to cycle through numbers and only goes up to 3), and it's not always obvious that this is a DFHack feature.

Reply:
That's right - the variable ui for setting wheelbarrows is the whole reason for this feature. This will set the wheelbarrow (and/or container) count programmatically